.github/workflows/node-js-ci.yml

```diff
@@ -19,13 +19,36 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
```

Comment:
We don't use `package-lock.json` in most cases.  What does `${{ hashFiles('**/package-lock.json') }}` do and can we live without it?

Reply:
we recache the node modules when this file changed.

Reply:
Right, but when we don't have it?